pkg/client/client.go

```diff
@@ -252,59 +253,54 @@ func (c Client) GetProviderConfiguration(ctx context.Context, providerName strin
 	return providerPlugin.Provider().GetProviderConfig(ctx, &cqproto.GetProviderConfigRequest{})
 }
 
-func (c Client) ExecutePolicy(ctx context.Context, request ExecutePolicyRequest) (*PolicyExecutionResult, error) {
+func (c Client) DownloadPolicy(ctx context.Context, args []string) error {
+	c.Logger.Info("Downloading policy from GitHub", "args", args)
+	m := policy.NewManager(c.PolicyDirectory, c.pool)
 
-	conn, err := c.pool.Acquire(ctx)
-	if err != nil {
-		return nil, err
-	}
-	defer conn.Release()
-	data, err := afero.ReadFile(afero.NewOsFs(), request.PolicyPath)
+	// Parse input args
+	p, err := m.ParsePolicyHubPath(args, "")
 	if err != nil {
-		return nil, err
+		return err
 	}
+	c.Logger.Debug("Parsed policy download input arguments", "policy", p)
+	return m.DownloadPolicy(ctx, p)
+}
 
-	var policy config.Policy
-	// TODO: convert to hcl
-	if err := yaml.Unmarshal(data, &policy); err != nil {
-		return nil, err
-	}
-	// Create Views
-	c.Logger.Debug("creating policy views", "policy", request.PolicyPath)
-	if err := createViews(ctx, conn, policy.Views); err != nil {
-		return nil, fmt.Errorf("failed to create policy views %w", err)
+func (c Client) RunPolicy(ctx context.Context, req PolicyRunRequest) error {
+	c.Logger.Info("Running policy", "args", req.Args)
+	m := policy.NewManager(c.PolicyDirectory, c.pool)
```

Comment:
It wouldn't! The policy managers functions are thread-safe. However, I agree that it may makes sense to document that in the code so future developers are aware of that.

Reply:
Decided to make the policy manager not a singleton anymore which also fixes the executor singleton issue.